src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+
+        for (size_t i : {1, 3, 5, 7})
+            weights[i] = weight_vals[ColorIndex::GREEN];
+
+        Dispatcher::set_deltas(deltas, delta_vals, root);
+    }
+
+protected:
+    enum ColorIndex : uint8_t { RED = 0, GREEN, BLUE };
+    constexpr static float delta_vals_[]  = {-1, 0, 1};
+    constexpr static float weight_vals_[] = {
+        0.25,   // RED
+        0.125,  // GREEN
+        0.01625 // BLUE
+    };
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<4>, Color::PURPLE>
+    : public ASplitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<4>>
+{
+    using Dispatcher = SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<4>>;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_val_, deltas, delta_val_);
+    }
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const& weight_val, //
+                      Deltas& deltas, float const& delta_val)
+    {
+        Dispatcher::set_weights(weights, weight_val);
+        Dispatcher::set_deltas(deltas, delta_val);
+    }
+
+protected:
+    constexpr static float delta_val_  = 0.571783;
+    constexpr static float weight_val_ = 0.25;
+};
+
+
+template</*default pattern*/>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<4>>
+    : public Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<4>, Color::PURPLE>
+{
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>, Color::PINK, Color::PURPLE>
+    : public ASplitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>>
+{
+    using Dispatcher
+        = SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Square>;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_vals_, deltas, delta_vals_);
+    }
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const weight_vals[], //
+                      Deltas& deltas, float const delta_vals[])
+    {
+        Dispatcher::set_weights(weights, weight_vals);
+        Dispatcher::set_deltas(deltas, delta_vals);
+    }
+
+protected:
+    constexpr static float delta_vals_[]  = {0.700909, 1.05786};
+    constexpr static float weight_vals_[] = {0.179488, 0.070512};
+};
+
+
+template</*default pattern*/>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>>
+    : public Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>, Color::PINK,
+                      Color::PURPLE>
```

Comment:
we already have this one just above no ?

Reply:
This is to expose a default type